tv-show/src/components/features/review/ReviewForm/ReviewForm.tsx

```diff
@@ -0,0 +1,41 @@
+import { IReview } from "@/typings/Review.type";
+import { getEmailAndAvatar } from "@/utils/randomUserGenerator";
+import { Button, Card, CardBody, Container, Flex, NumberInput, NumberInputField, Textarea } from "@chakra-ui/react";
+import { useState } from "react";
+import Rating from "../../rating/Rating/Rating";
+
+interface IReviewForm {
+  onSubmitClick: (newReview: IReview) => void;
+}
+
+export default function ReviewForm({onSubmitClick}: IReviewForm) {
+  const [comment, setComment] = useState('');
+  const [rating, setRating] = useState(0);
+
+  const { email, avatar } = getEmailAndAvatar();
+
+  return(
+    <Container marginBottom={4}>
+      <Card>
+        <CardBody>
+          <Textarea
+            placeholder='What did you think of the show?'
+            onChange={(e) => setComment(e.target.value)}
+          />
+          <Rating label='rating-label' onChange={(n: number) => setRating(n)} value={rating}/>
+          <Flex flexDir='row-reverse'>
+            <Button onClick={() => {
+              const newReview: IReview = {
+                comment,
+                rating,
+                email,
+                avatar
+              };
+              onSubmitClick(newReview);
```

Comment:
NItpicking, but naming-wise, this arrow function is the onSubmitClick handler/function. The one you are getting from the props would be `onSubmit`, `onPostReview` or something similar. Whether that happened onClick or on keyPress or on whatever doesn't matter.